src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -961,7 +980,7 @@ ReorderInsertSelectTargetLists(Query *originalQuery, RangeTblEntry *insertRte,
 		 */
 		Assert(!newSubqueryTargetEntry->resjunk);
 
-		Var *newInsertVar = makeVar(insertTableId, originalAttrNo,
+		Var *newInsertVar = makeVar(insertTableId, newSubqueryTargetEntry->resno,
```

Comment:
Here resno should point to the attno of the subquery's relevant target list entry (as the other fields).
There is also that comment postgres codebase:
```c
 * In a SELECT's targetlist, resno should always be equal to the item's
 * ordinal position (counting from 1).  However, in an INSERT or UPDATE
 * targetlist, resno represents the attribute number of the destination
 * column for the item; so there may be missing or out-of-order resnos.
```

TODO: understand why this wasn't causing any issues before, e.g.: for non-pushable inserts

Reply:
Even more, we should simply use makeVarFromTargetEntry() after resolving above^ comment.

Reply:
similar to https://github.com/citusdata/citus/pull/5873#discussion_r841525833